clang/lib/Sema/Sema.cpp

```diff
@@ -988,12 +988,13 @@ void Sema::ActOnEndOfTranslationUnitFragment(TUFragmentKind Kind) {
     PerformPendingInstantiations();
   }
 
-  if (getLangOpts().SYCLIsDevice) {
+  if (getLangOpts().SYCLIsDevice || getLangOpts().SYCLIsHost) {
```

Comment:
I think you enabled this for the host, so that MarkDevice could be called.   Correct?   But it also makes IntHeader emitted on host; I don't think we need that.   I think we should split this; suggest

```
if (getLangOpts().SYCLIsDevice)
  if (SyclIntHeader != nullptr)
     SyclIntHeader->emit(getLangOpts().SYCLIntHeader);
if (getLangOpts().SYCLIsDevice || getLangOpts().SYCLIsHost)
   MarkDevice();
```

Reply:
Yes, correct. I missed the fact that we do not need to emit IntHeader on host. I have updated patch.